product-mini/platforms/zephyr/simple/README.md

```diff
@@ -1,106 +1,82 @@
+the expense of building a rather large image, which both can take a long time to
+build and uses up a large amount of storage (~15 GB).
 
-Adopt the device or remove if not needed.
+Execute the following command to build the Docker image. This may take an
+extended period of time to complete.
 
 ```shell
-docker run -ti --device=/dev/ttyUSB0 wamr-zephyr
-```
-
-And then inside the docker container:
-
-```shell
-# copy the corresponding board conf file to current directory
-cp boards/qemu_x86_nommu.conf prj.conf
-# then build
-./build_and_run.sh x86
+docker build -t wamr-zephyr .
 ```
 
-> PS: for boards esp32, need to configure some environment first
+Execute the following command to run the image built in the previous step. If
+you are planning to flash a device after building, make sure to specify it with
+[`--device`](https://docs.docker.com/engine/reference/run/#runtime-privilege-and-linux-capabilities).
 
 ```shell
-# configure zephyr with espressif
-export ZEPHYR_TOOLCHAIN_VARIANT="espressif"
-export ESPRESSIF_TOOLCHAIN_PATH="/root/.espressif/tools/xtensa-esp32-elf/esp-2019r2-8.2.0/xtensa-esp32-elf/"
-export ESP_IDF_PATH="/root/esp/esp-idf"
-# copy the corresponding board conf file to current directory
-cp boards/esp32.conf prj.conf
-# then build
-./build_and_run.sh esp32
+docker run -it --rm --device=/dev/ttyUSB0 wamr-zephyr
 ```
 
-## Build on local environment
-
-### Dependencies installation
+### Local Environment
 
-Following the Zephyr and Espressif official document:
+Zephyr can also be setup locally to enable building this sample application.
+This allows you have have more control over what modules and tools are
+installed, which can drastically reduce the storage required compared to the
+Docker image.
 
-1. Zephyr installation:
+Follow the steps provided in the [Zephyr Getting Started
+guide](https://docs.zephyrproject.org/latest/develop/getting_started/index.html)
+to setup for local development.
 
-   <https://docs.zephyrproject.org/latest/develop/getting_started/index.html>
+## Building for a Specific Board
 
-2. ESP32 installation:
-
-   <https://docs.espressif.com/projects/esp-idf/en/latest/esp32/get-started/linux-macos-setup.html>
-
-And setup the Zephyr for esp32:
-
-<https://wiki.amarulasolutions.com/zephyr/esp32/esp32-setup.html>
-
-Then Installing QEMU, for example, on Linux:
+With an environment setup either locally or in a Docker container, you can build
+for a Zephyr suppported board using
+[`west`](https://docs.zephyrproject.org/latest/develop/west/index.html). There
+are already [configuaration files](./boards) for a few boards in this sample.
+However, if you are using a new board, you will need to add your own file for
+the board, or define configuration in the [`prj.conf](./prj.conf). After doing
+so, use the following command with your board identifier to build the sample
+application.
 
 ```shell
-sudo apt-get install qemu
+west build . -b <board-identifier> --pristine -- -DWAMR_BUILD_TARGET=<wamr-arch>
```

Comment:
Maybe it's better to add a step to copy the `boards/<board>.conf` to `prj.conf` so that a newbie like me won't forget? 😄

Reply:
Ah great catch! I meant to just had an empty `prj.conf`, which should alleviate this issue :) I may also move some of the config that is not specific to a single board into the `prj.conf` 👍🏻